### ISSUE TITLE
[BUG] Import File Activity Missing

### ISSUE THREAD

**Issue by solariimeow** [Accenture/sfmc-devtools#198]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Getting an error with import file activity blocks the entire process of retrieval.
<img width="692" alt="Import File Activity Error" src="https://user-images.githubusercontent.com/45225495/155295916-915e39a8-845a-496e-af17-4997b18123e5.PNG">

Import file activity exists in MC environment.
<img width="884" alt="Import File Activity" src="https://user-images.githubusercontent.com/45225495/155295267-fcf052c2-ba81-4428-ade4-2c4cded153cf.PNG">

### Expected Behavior

Be able to retrieve the files successfully.

### Steps To Reproduce

1. Extract the automation from the source BU with retrieve command.
2. Error occurs immediately.

### Version

3.1.3

### Environment

- OS: Windows 10 Enterprise
- Node: 16.13.0
- npm: 8.4.0

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by JoernBerkefeld** [comment]
@DougMidgley  - did you encounter this before?

**Comment by JoernBerkefeld** [comment]
@solariimeow could you please try `mcdev retrieve . importFile` to check if the one causing this is downloaded? 

please also attach the log file that the command will create here

**Comment by DougMidgley** [comment]
Hello, we have seen some issues with this before with old activities, can you confirm if this is in Automation Studio visible (not via Automations), and also can you provide when this was actually created?

**Comment by solariimeow** [comment]
Hello @JoernBerkefeld , please see attached for the log file. :)
[2022-02-23T09.39.31.164Z.log](https://github.com/Accenture/sfmc-devtools/files/8123233/2022-02-23T09.39.31.164Z.log)

Hiya @DougMidgley , the sample automation was created on 08/26/2021 11:30 PM - and yes, it's visible in Automation Studio.

**Comment by JoernBerkefeld** [comment]
the issue is not about the automation itself but the importFile activity. if that's an old one, then what Doug suggested could be the root cause.

checking the log now

**Comment by solariimeow** [comment]
Hello~ What would be the best course of action here? We're experiencing the same issue with a few automations too. Thank you.

**Comment by solariimeow** [comment]
**Error messages:** @JoernBerkefeld @DougMidgley 

**SBG_BGUS_C2E MyCTL Migration**
Import file created on 11/02/2021 2:35 AM

`Missing importFile activity 'SBG_BGUS_C2E MyCTL Migration' in step 1.1 of Automation 'SBG_BGUS_C2E MyCTL Migration' (Missing one or more dependent metadata. importFile with importDefinitionId='a5d64c81-423b-ec11-b82c-48df37e41f4c' was not found on your BU)`
Automation Studio:
<img width="599" alt="SBG_BGUS_C2E MyCTL Migration" src="https://user-images.githubusercontent.com/45225495/155520980-257324e3-d48f-41da-8b69-5085ec3f9f36.PNG">
Import File:
<img width="922" alt="SBG_BGUS_C2E MyCTL Migration Import File" src="https://user-images.githubusercontent.com/45225495/155521037-f4fd6402-b7ef-41fa-bbeb-3126c2898c0e.PNG">

**SBG_BGUS_C2E_Pre Migration 1st Touch**
Import file created on 08/26/2021 11:30 PM

`Missing importFile activity 'SBG_BGUS_C2E_Pre Migration 1st Touch' in step 1.1 of Automation 'SBG_BGUS_C2E_Pre Migration 1st Touch' (Missing one or more dependent metadata. importFile with importDefinitionId='32d47493-8206-ec11-b82c-48df37e41d3a' was not found on your BU)`
Automation Studio:
<img width="595" alt="SBG_BGUS_C2E_Pre Migration 1st Touch" src="https://user-images.githubusercontent.com/45225495/155521174-ce144cde-37c2-497a-978d-087d07789951.PNG">
Import File:
<img width="933" alt="SBG_BGUS_C2E_Pre Migration 1st Touch Import File" src="https://user-images.githubusercontent.com/45225495/155521248-353059d1-8f32-4e2f-b635-6d2aacdb3100.PNG">

Please let me know if you need anything else. :)

**Comment by JoernBerkefeld** [comment]
hm strange. when you just try to retrieve the importFile activities - does that work for those 2? do they show up in your BUs "importFile" subfolder?

**Comment by solariimeow** [comment]
Unfortunately it's not showing anywhere at all. I looked up the import file external key (**4FA8C36F-B66D-412B-AE41-3CB254764679**) from SBG_BGUS_C2E MyCTL Migration and it didn't show in importFile subfolder, and error message didn't display anything related to it. Please see below log file. Thank you.
[2022-02-28T13.49.38.654Z.log](https://github.com/Accenture/sfmc-devtools/files/8153822/2022-02-28T13.49.38.654Z.log)

**Comment by JoernBerkefeld** [comment]
@DougMidgley you are our "this should be working but it doesn't"-expert ;-)

**Comment by JoernBerkefeld** [comment]
your log states that 50 import file activities should have been downloaded - of which 33 had problems concerning finding the linked DataExtension.

**Comment by DougMidgley** [comment]
Indeed looks like the Dependent Data Extensions are missing, which means Import Activities are, which is showing up in Automations. 
On a side note: its worth double checking any content that you share that you are ok with the content, and nothing senstive is exposed. Also, we consider best practise to have Name and External Key/Customer Key matching. This is because we use the external key to keep uniqueness, and therefore it is filename. By naming both the same it is easier for people to quickly see in the directory which files are being looked at:)

**Comment by JoernBerkefeld** [comment]
If it was that then the 4.0.0 should at least resolve the retrieve issue and then ease debugging. We changed errors to warnings during retrieve to ensure we can download everything even with missing deps

**Comment by jnguyenandersen** [comment]
I'm having a similar issue to this, this is the message I was getting (stripped out information specific to my instance)
```
error: Missing importFile activity 'NAME_HERE' in step 1.1 of Automation 'AUTOMATION_NAME_HERE' (Missing one or more dependent metadata. importFile with importDefinitionId='########-d8b7-ea11-a2f7-##########' was not found on your BU)
```
I checked the UI and the import exists, and it's also running successfully on a daily basis so the error message isn't quite lining up.

**Comment by JoernBerkefeld** [comment]
@jnguyenandersen - could you please try the current developer build to check if the error persists?

To install, run 

npm -i -g accenture/mcdev#develop

That installs directly from our develop branch here on github. With a bit of luck we fixed that

**Comment by JoernBerkefeld** [comment]
related discussion: https://github.com/Accenture/sfmc-devtools/discussions/482

**Comment by JoernBerkefeld** [comment]
at least some issues might be resolved with the new 5.0 release.
Please let us know if you continue to have issues retrieving importFile activities.
